Controller/CRUDController.php

```diff
@@ -221,7 +221,8 @@ public function deleteAction($id)
     /**
      * Edit action.
      *
-     * @param int|string|null $id Optional ID to use instead of globally provided default.
+     * @param int|string|null $id Optional ID representing the main admin subject.
```

Comment:
With a child admin the url will be ``/author/ID/post/CHILD_ID``. The Admin Bundle will automagically filter post by author's id on the list mode or set the author when a post is created (ie ``/author/ID/post/create``). 

The AdminBundle will start the same controller for ``/post/ID`` or ``/author/ID/post/CHILD_ID``.

The method ``getIdParameter`` find the correct reference for the current admin.

I cannot remember why we add ``$id`` in the action method. This feature is part of the core of sonata, and it will be hard to refactor. However, once it will be done, you will get a pretty good insight of the code ;)